lib/src/annotation.dart

```diff
@@ -150,7 +150,17 @@ dynamic _createFromConstructor(
       // field assigned in the object. Then we can take the field value and
       // set it as the argument value
 
+      // If the annotation class's constructor has no initializers and just
+      // forwards to a `super` class constructor, then there is no need
+      // to add a ctor argument name as a field value.
+      //
+      // In this case `singleWhere` would throw a StateError.
+      if (ctor.constantInitializers.length == 1 && ctor.constantInitializers.first is SuperConstructorInvocation)
+        continue;
```

Comment:
This happens in a case like the following:

```dart
// Relationship is also an annotation class
class HasMany extends Relationship {
 // This `super` call 
 const HasMany():super();
}
```

`ctor.constructorInitializers` will contain a `SuperConstructorInvocation`, which throws an error because we try to cast it as `ConstructorFieldInitializer`.

I added a test involving this, but perhaps I wasn't clear about which it was:
```dart
test('annotated with extended class', () {
      var instance = _getInstantiatedAnnotation(libElement, 'WithChildClassAnnotation');
      expect(instance is defs.PublicAnnotationClass, isTrue);
      expect(instance is defs.ExtendedAnnotationClass, isTrue);
      expect(instance.anInt, 0);
      expect(instance.aBool, isFalse);
    });

// annotated_classes.dart
@ExtendedAnnotationClass()
class WithChildClassAnnotation {}

// annotations.dart
class ExtendedAnnotationClass extends PublicAnnotationClass {
  const ExtendedAnnotationClass():super();
}
```

Also, yes, it seems kind of iffy to skip a parameter, but it would always throw an error because the code would cast each member of `constructorInitalizers` to `ConstructorFieldInitializer`, which would always throw a type error. I'm hoping there is a better way to do this, but this seems to work.